packages/Modal/src/Modal.js

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import extractChildren from "@paprika/helpers/lib/extractChildren";
+import * as styled from "./Modal.styles";
+
+const propTypes = {
+  /** The content for the Modal. */
+  children: PropTypes.node.isRequired,
+
+  /** Control the visibility of the modal */
+  isOpen: PropTypes.bool.isRequired,
+
+  /** Callback triggered when the modal needs to be close */
+  onClose: PropTypes.func,
+
+  /** Callback once the modal has been opened event */
+  onAfterOpen: PropTypes.func,
+
+  /** Callback once the modal has been closed event */
+  onAfterClose: PropTypes.func,
+
+  /* Control the size (max-width) of the modal */
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+};
+
+const defaultProps = {
+  onAfterClose: () => {},
+  onClose: () => {},
+  onAfterOpen: () => {},
+  size: ShirtSizes.MEDIUM,
+};
+
+const Modal = props => {
+  const { isOpen, onClose, onAfterClose, onAfterOpen, size, ...moreProps } = props;
+
+  const {
+    "Modal.FocusTrap": focusTrapExtracted,
+    "Modal.Header": headerExtracted,
+    "Modal.Content": contentExtracted,
+    "Modal.Footer": footerExtracted,
+    children,
+  } = extractChildren(moreProps.children, ["Modal.FocusTrap", "Modal.Header", "Modal.Content", "Modal.Footer"]);
+
+  const extendedFocusTrapOptions = focusTrapExtracted ? focusTrapExtracted.props : {};
+
+  return (
+    <styled.Overlay
+      isOpen={isOpen}
+      onClose={onClose}
+      onAfterOpen={onAfterOpen}
+      onAfterClose={onAfterClose}
+      focusTrapOptions={extendedFocusTrapOptions}
+    >
+      {state => (
+        <styled.Wrapper size={size}>
+          <styled.Dialog state={state} role="dialog" data-pka-anchor="modal">
+            {headerExtracted && <styled.Header {...headerExtracted.props} onClose={onClose} />}
+            <styled.ContentWrapper role="region" tabIndex="0">
```

Comment:
I don't think this should have a tabIndex of `0` – it is not interactive and doesn't need to be in the keyboard tab order.  

Does it need a `-1` so that it can be the fallback focussable element on open?

Reply:
`tabIndex=0` adds ability to focus this area. It is required to allow scroll this area by keyboard. Without this user don't have an ability scroll content area by keyboard.

Reply:
This is a good point, and a bit of a blind spot in the accessibility material I've been studying.  Let's leave it like this for now, but I've asked @sunminsarahlee  to mock up an alternative focus style for this kind of situation where the focussed element is _NOT_ actionable, which we can apply in a follow up task.